cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -100,20 +104,44 @@ func MustNew(cfg config.LocalConfig) *Daemon {
 	if len(cfg.HistoryArchiveURLs) == 0 {
 		logger.Fatalf("no history archives url were provided")
 	}
-	historyArchive, err := historyarchive.Connect(
-		cfg.HistoryArchiveURLs[0],
-		historyarchive.ConnectOptions{
-			CheckpointFrequency: cfg.CheckpointFrequency,
+	// Defaults: https://pkg.go.dev/github.com/cenkalti/backoff/v4@v4.2.0#pkg-constants
```

Comment:
I think for db and archive resources, the backoff may not be adding much value and not worth the xtra code to wrap it as such.  both of those resources are either on/off, they don't have transition states, I think fast failing on both of these is fine still as that means it's likely a configuration error, @tamirms do you agree?

Reply:
I think that as a service, this back off initialization isn't appropriate.
We should startup fast ( i.e. less than 2 sec ), and fail fast.
Any network activity should follow later and keep trying to recover forever

Reply:
I.e. it's a server. There is no one behind the console. If it can start up, it should endlessly try to take care of itself.
If it's not completely initialized, it should report that to the various endpoints.

Reply:
The reason I added retrying with back-off to the archive here is because #358 suggested it as a possible solution. From my understanding, connecting to the history archive may fail for some network related reason, and retrying would help with this. However, based on the comments in this PR, it seems like it is only maybe worth retrying running ingestion.

This makes me wonder: **what exactly are the possible transient core/network state issues we want to make more resilient?** It seems I'm misunderstanding something important here, or there are very little/no transient core/network state issues to begin with. Would appreciate any insight here.

Reply:
I think published archives are one transient state to retry, the bug report on #358 , referred to a stack trace related to rpc exiting early when it tries to do ingestion system startup due to the ingest library throwing error when not able to handle empty archive service, it expected at least one archive be published by the service, however in case of starting a new networks like standalone, the archives http service will be up, but no archives are published yet until the first checkpoint occurs on the network(once every 64 ledgers, or 8 ledgers if using accelerated mode). The error from ingest library worked up stack into rpc service initialization routine and it panic'd, causing rpc service to exit, however, In 5 minutes(or sooner on accelerated network) that archive state will change as the first 64 ledgers complete and the checkpoint archive will be published at which point the ingest routine can proceed, so a retry from rpc over that initialization routine and timeframe should prevent a panic, allowing the same rpc service to continue and recover from that type of evolving network state.

The retry applied on https://github.com/stellar/soroban-tools/pull/543/files#diff-b8bd4f66b43a1d0998b64e915e915f36752c2d8d8a50f46ed74f27f92f92eb4aR56 covers the rpc starting the ingestion system, so, seems like it would be effective in this regard.

There were some recent changes in the ingestion library, [go#4773](https://github.com/stellar/go/pull/4773) to be more robust in handling some aspects of ledger ranges based on archives, but I'm not sure those changes prevent the error due to no-archives present case. It would be good to replicate the bug on present day source, since #358 was reported several weeks ago, by running a standalone network from `stellar/quickstart:soroban-dev`, and check the supervisor rpc logs in the running container to see if rpc process is still exiting early and supervisor is restarting it repeatedly.

Reply:
I learned a lot from these comments. Thank you for the detailed responses! I removed retrying connecting to the history archives here since it's unnecessary.

I'll figure out unit testing retries for running ingestion in order to finalize this PR.